File: App/Models/Product.cs
using System;

namespace Rbyte.App.Models
{
    public class Product
    {
        public int ProductId { get; private set; }
        public string Name { get; private set; }
        public string Description { get; private set; }
        public decimal StandardPrice { get; private set; }
        public string Producent { get; private set; }

        // constructor
        public Product(string name, string description, decimal standardPrice, string producent)
        {
            ProductId = new Random().Next();
            Name = name;
            Description = description;
            StandardPrice = standardPrice;
            Producent = producent;
        }
    }
}

File: Rbyte/Src/Presentation/Rbyte.Angular/rbyte/src/app/product/models/product.ts
export class Product {
    constructor(productId: number, name: string,
        description: string, barcode: number,
        tax: number, priceWithoutMargin: number,
        fullPrice: number) {
        this.productId = productId;
        this.name = name;
        this.description = description;
        this.barcode = barcode;
        this.tax = tax;
        this.priceWithoutMargin = priceWithoutMargin;
        this.fullPrice = fullPrice;
    }
    productId: number;
    name: string;
    description: string;
    barcode: number;
    tax: number;
    priceWithoutMargin: number;
    fullPrice: number;
}
